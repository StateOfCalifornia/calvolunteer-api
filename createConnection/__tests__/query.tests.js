const query = require('../query');

test('createQuery should return graphql', () => {
    var params = {
        oppId: 1234,
        replies: [{
            id: 123,
            values: ['a', 'b']
        }],
        email: 'a@b.com',
        firstName: 'first',
        lastName: 'last',
        phoneNumber: "123-456-7890",
        zipCode: '12345',
        acceptTermsAndConditions: true,
    }
    var graphql = query.generateQuery(params);
    expect(graphql).toMatchSnapshot();
})

test('createQuery should return graphql with no replies set', () => {
    var params = {
        oppId: 1234,
        replies: 0,
        email: 'a@b.com',
        firstName: 'first',
        lastName: 'last',
        phoneNumber: "123-456-7890",
        zipCode: '12345',
        acceptTermsAndConditions: true,
    }
    var graphql = query.generateQuery(params);
    expect(graphql).toMatchSnapshot();
})