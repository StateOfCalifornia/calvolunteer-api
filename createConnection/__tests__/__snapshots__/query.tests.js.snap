// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createQuery should return graphql 1`] = `
"mutation {
        createConnection ( 
            input: {
            oppId: 1234
            replies: [{ id: 123, values: [\\"a\\",\\"b\\"] },]
            comments: \\"Connection created by #CaliforniansForAll\\"
            volunteer: {
                email: \\"a@b.com\\"
                firstName: \\"first\\"
                lastName: \\"last\\"
                phoneNumber: \\"123-456-7890\\"
                zipCode: \\"12345\\"
                acceptTermsAndConditions: true
            } 
            }) 
            {
            oppId
            comments
            volunteer {
                email
                firstName
                lastName
                phoneNumber
                zipCode
            }
            enlister {
                email
                firstName
                lastName
                phoneNumber
                zipCode
            }
            shifts {
                id
                name
                notes
                date
                startTime
                endTime
                volNeeded
            }
            replies {
                id
                values
            }
        }
    }"
`;

exports[`createQuery should return graphql with no replies set 1`] = `
"mutation {
        createConnection ( 
            input: {
            oppId: 1234
            replies: []
            comments: \\"Connection created by #CaliforniansForAll\\"
            volunteer: {
                email: \\"a@b.com\\"
                firstName: \\"first\\"
                lastName: \\"last\\"
                phoneNumber: \\"123-456-7890\\"
                zipCode: \\"12345\\"
                acceptTermsAndConditions: true
            } 
            }) 
            {
            oppId
            comments
            volunteer {
                email
                firstName
                lastName
                phoneNumber
                zipCode
            }
            enlister {
                email
                firstName
                lastName
                phoneNumber
                zipCode
            }
            shifts {
                id
                name
                notes
                date
                startTime
                endTime
                volNeeded
            }
            replies {
                id
                values
            }
        }
    }"
`;
